frontend/common/APICall.js

```diff
@@ -0,0 +1,77 @@
+import { Links } from "./constants.js";
+class APICall{
+
+    constructor(){
+        this.authToken = localStorage.getItem('authToken');
+        this.headers = new Headers({
+            'Authorization': `Bearer ${this.authToken}`,
+            'Content-Type': 'application/json',
+        });
+    }
+    
+    makeGetRequest(path) {
```

Comment:
May we please name this function to something a bit more descriptive, "makeGetRequest" is ambiguous and doesn't show what the request actually is.

Reply:
I think the name is fine because it is a generic method to make any get request...
But you can suggest any name then I'll rename it

Reply:
I think the name is also fine